// @aws-sdk/client-iam/dist-es/models/models_1.js
import { __assign, __extends } from "tslib";
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
import { IAMServiceException as __BaseException } from "./IAMServiceException";
export var UntagPolicyRequest;
(function (UntagPolicyRequest) {
    UntagPolicyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagPolicyRequest || (UntagPolicyRequest = {}));
export var UntagRoleRequest;
(function (UntagRoleRequest) {
    UntagRoleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagRoleRequest || (UntagRoleRequest = {}));
export var UntagSAMLProviderRequest;
(function (UntagSAMLProviderRequest) {
    UntagSAMLProviderRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagSAMLProviderRequest || (UntagSAMLProviderRequest = {}));
export var UntagServerCertificateRequest;
(function (UntagServerCertificateRequest) {
    UntagServerCertificateRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagServerCertificateRequest || (UntagServerCertificateRequest = {}));
export var UntagUserRequest;
(function (UntagUserRequest) {
    UntagUserRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagUserRequest || (UntagUserRequest = {}));
export var UpdateAccessKeyRequest;
(function (UpdateAccessKeyRequest) {
    UpdateAccessKeyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateAccessKeyRequest || (UpdateAccessKeyRequest = {}));
export var UpdateAccountPasswordPolicyRequest;
(function (UpdateAccountPasswordPolicyRequest) {
    UpdateAccountPasswordPolicyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateAccountPasswordPolicyRequest || (UpdateAccountPasswordPolicyRequest = {}));
export var UpdateAssumeRolePolicyRequest;
(function (UpdateAssumeRolePolicyRequest) {
    UpdateAssumeRolePolicyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateAssumeRolePolicyRequest || (UpdateAssumeRolePolicyRequest = {}));
export var UpdateGroupRequest;
(function (UpdateGroupRequest) {
    UpdateGroupRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateGroupRequest || (UpdateGroupRequest = {}));
export var UpdateLoginProfileRequest;
(function (UpdateLoginProfileRequest) {
    UpdateLoginProfileRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Password && { Password: SENSITIVE_STRING }))); };
})(UpdateLoginProfileRequest || (UpdateLoginProfileRequest = {}));
export var UpdateOpenIDConnectProviderThumbprintRequest;
(function (UpdateOpenIDConnectProviderThumbprintRequest) {
    UpdateOpenIDConnectProviderThumbprintRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateOpenIDConnectProviderThumbprintRequest || (UpdateOpenIDConnectProviderThumbprintRequest = {}));
export var UpdateRoleRequest;
(function (UpdateRoleRequest) {
    UpdateRoleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateRoleRequest || (UpdateRoleRequest = {}));
export var UpdateRoleResponse;
(function (UpdateRoleResponse) {
    UpdateRoleResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateRoleResponse || (UpdateRoleResponse = {}));
export var UpdateRoleDescriptionRequest;
(function (UpdateRoleDescriptionRequest) {
    UpdateRoleDescriptionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateRoleDescriptionRequest || (UpdateRoleDescriptionRequest = {}));
export var UpdateRoleDescriptionResponse;
(function (UpdateRoleDescriptionResponse) {
    UpdateRoleDescriptionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateRoleDescriptionResponse || (UpdateRoleDescriptionResponse = {}));
export var UpdateSAMLProviderRequest;
(function (UpdateSAMLProviderRequest) {
    UpdateSAMLProviderRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateSAMLProviderRequest || (UpdateSAMLProviderRequest = {}));
export var UpdateSAMLProviderResponse;
(function (UpdateSAMLProviderResponse) {
    UpdateSAMLProviderResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateSAMLProviderResponse || (UpdateSAMLProviderResponse = {}));
export var UpdateServerCertificateRequest;
(function (UpdateServerCertificateRequest) {
    UpdateServerCertificateRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateServerCertificateRequest || (UpdateServerCertificateRequest = {}));
export var UpdateServiceSpecificCredentialRequest;
(function (UpdateServiceSpecificCredentialRequest) {
    UpdateServiceSpecificCredentialRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateServiceSpecificCredentialRequest || (UpdateServiceSpecificCredentialRequest = {}));
export var UpdateSigningCertificateRequest;
(function (UpdateSigningCertificateRequest) {
    UpdateSigningCertificateRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateSigningCertificateRequest || (UpdateSigningCertificateRequest = {}));
export var UpdateSSHPublicKeyRequest;
(function (UpdateSSHPublicKeyRequest) {
    UpdateSSHPublicKeyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateSSHPublicKeyRequest || (UpdateSSHPublicKeyRequest = {}));
export var UpdateUserRequest;
(function (UpdateUserRequest) {
    UpdateUserRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateUserRequest || (UpdateUserRequest = {}));
var KeyPairMismatchException = (function (_super) {
    __extends(KeyPairMismatchException, _super);
    function KeyPairMismatchException(opts) {
        var _this = _super.call(this, __assign({ name: "KeyPairMismatchException", $fault: "client" }, opts)) || this;
        _this.name = "KeyPairMismatchException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, KeyPairMismatchException.prototype);
        return _this;
    }
    return KeyPairMismatchException;
}(__BaseException));
export { KeyPairMismatchException };
var MalformedCertificateException = (function (_super) {
    __extends(MalformedCertificateException, _super);
    function MalformedCertificateException(opts) {
        var _this = _super.call(this, __assign({ name: "MalformedCertificateException", $fault: "client" }, opts)) || this;
        _this.name = "MalformedCertificateException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, MalformedCertificateException.prototype);
        return _this;
    }
    return MalformedCertificateException;
}(__BaseException));
export { MalformedCertificateException };
export var UploadServerCertificateRequest;
(function (UploadServerCertificateRequest) {
    UploadServerCertificateRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.PrivateKey && { PrivateKey: SENSITIVE_STRING }))); };
})(UploadServerCertificateRequest || (UploadServerCertificateRequest = {}));
export var UploadServerCertificateResponse;
(function (UploadServerCertificateResponse) {
    UploadServerCertificateResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UploadServerCertificateResponse || (UploadServerCertificateResponse = {}));
var DuplicateCertificateException = (function (_super) {
    __extends(DuplicateCertificateException, _super);
    function DuplicateCertificateException(opts) {
        var _this = _super.call(this, __assign({ name: "DuplicateCertificateException", $fault: "client" }, opts)) || this;
        _this.name = "DuplicateCertificateException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DuplicateCertificateException.prototype);
        return _this;
    }
    return DuplicateCertificateException;
}(__BaseException));
export { DuplicateCertificateException };
var InvalidCertificateException = (function (_super) {
    __extends(InvalidCertificateException, _super);
    function InvalidCertificateException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidCertificateException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidCertificateException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidCertificateException.prototype);
        return _this;
    }
    return InvalidCertificateException;
}(__BaseException));
export { InvalidCertificateException };
export var UploadSigningCertificateRequest;
(function (UploadSigningCertificateRequest) {
    UploadSigningCertificateRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UploadSigningCertificateRequest || (UploadSigningCertificateRequest = {}));
export var UploadSigningCertificateResponse;
(function (UploadSigningCertificateResponse) {
    UploadSigningCertificateResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UploadSigningCertificateResponse || (UploadSigningCertificateResponse = {}));
var DuplicateSSHPublicKeyException = (function (_super) {
    __extends(DuplicateSSHPublicKeyException, _super);
    function DuplicateSSHPublicKeyException(opts) {
        var _this = _super.call(this, __assign({ name: "DuplicateSSHPublicKeyException", $fault: "client" }, opts)) || this;
        _this.name = "DuplicateSSHPublicKeyException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DuplicateSSHPublicKeyException.prototype);
        return _this;
    }
    return DuplicateSSHPublicKeyException;
}(__BaseException));
export { DuplicateSSHPublicKeyException };
var InvalidPublicKeyException = (function (_super) {
    __extends(InvalidPublicKeyException, _super);
    function InvalidPublicKeyException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidPublicKeyException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidPublicKeyException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidPublicKeyException.prototype);
        return _this;
    }
    return InvalidPublicKeyException;
}(__BaseException));
export { InvalidPublicKeyException };
export var UploadSSHPublicKeyRequest;
(function (UploadSSHPublicKeyRequest) {
    UploadSSHPublicKeyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UploadSSHPublicKeyRequest || (UploadSSHPublicKeyRequest = {}));
export var UploadSSHPublicKeyResponse;
(function (UploadSSHPublicKeyResponse) {
    UploadSSHPublicKeyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UploadSSHPublicKeyResponse || (UploadSSHPublicKeyResponse = {}));
